### PR TITLE
change notification library and also change notification fire call

### LINKED ISSUE #727: wallet crashed when receiving part
we were doing a test with @vikas-cis and he sent me 2 tPART, when I received it I seen the notification and the wallet crashed right away with this output on the terminal console : 

```
2018-01-19 16:19:36 connect() to 196.25.195.98:51938 failed after select(): Connection refused (111)
2018-01-19 16:19:39 AddToWallet adc9feb6aec404b123ef5beb0aa6b9ceac58fd08dda676c4f9c01a8948711788  new
[14229:0119/171940.754195:FATAL:wrappable.cc(27)] Check failed: !wrapper_.IsEmpty(). 
#0 0x000002bf6847 <unknown>
#1 0x000002bf4cfb <unknown>
#2 0x000003e91eca mate::WrappableBase::GetWrapper()
#3 0x000003ec5331 _ZN4mate12EventEmitterIN4atom3api12NotificationEE14EmitWithSenderIJEEEbRKN4base16BasicStringPieceISsEEPN7content11WebContentsEPN3IPC7MessageEDpRKT_
#4 0x000003ec2fc8 atom::api::Notification::NotificationDestroyed()
#5 0x000003fcc3e0 brightray::Notification::~Notification()
#6 0x000003fcc9a9 brightray::LibnotifyNotification::~LibnotifyNotification()
#7 0x000003fbcc10 brightray::NotificationPresenter::RemoveNotification()
#8 0x7f5006aed6f5 g_closure_invoke
#9 0x7f5006b010b0 <unknown>
#10 0x7f5006b05696 g_signal_emit_valist
#11 0x7f5006b06920 g_signal_emit
#12 0x7f4fd72c88bd <unknown>
#13 0x7f500000b1c8 ffi_call_unix64
#14 0x7f500000ac2a ffi_call
#15 0x7f5006af684b g_cclosure_marshal_generic
#16 0x7f5006aed6f5 g_closure_invoke
#17 0x7f5006b010b0 <unknown>
#18 0x7f5006b05696 g_signal_emit_valist
#19 0x7f5006b06920 g_signal_emit
#20 0x7f5006fdd31f <unknown>
#21 0x7f5007052ea9 <unknown>
#22 0x7f5003ec00be g_main_context_dispatch
#23 0x7f5003ec1f69 <unknown>
#24 0x7f5003ec1fae g_main_context_iteration
#25 0x000002be7e96 <unknown>
#26 0x000002c602d7 <unknown>
#27 0x000002c6db6e <unknown>
#28 0x000000eb2bcc <unknown>
#29 0x000000eb2a15 <unknown>
#30 0x000000ed2b3d <unknown>
#31 0x0000012f853e <unknown>
#32 0x000000c7905c <unknown>
#33 0x000000c794b0 <unknown>
#34 0x000003e3da88 main
#35 0x7f50017c3f4a __libc_start_main
#36 0x0000005b0e99 <unknown>

[1]    14229 abort (core dumped)  ./packages/linux-unpacked/particl
```

After restarting the wallet, daemon was still running from the previous session, so the crash is from the GUI, not the daemon.

### LINKED ISSUE #810: Switch Notification library
Swap out the Electron notify application out for another one.
We'll have to check what's available..

### PR DESCRIPTION
**Pull request by vikas-cis**
Fixes #810
Fixes #727

### CODE REVIEW

**Comment by vikas-cis** [comment]
screen shot 2018-02-09 at 6 48 36 pm

This is how it looks like in mac, I think for current scenario we can just make it as it is and once we have electron 1.8.2 stable version comes up then will revert that and used the default behaviour.

**Comment by pciavald** [comment]
@kewde where should this PR go ?

@HenkSwardt

**Comment by kewde** [comment]
Untested, but should go in 1.1.0

**Comment by vikas-cis** [comment]
@kewde which one will go node-notifier or electron-notify ?

**Comment by rynomster** [comment]
This doesn't look like it will make it into 1.1.0

**Comment by HenkSwardt** [comment]
Please confirm that this is needed for the Electron build. As far as I had it we needed this for the Muon conversion. The old Electron branch had a fully working notification system?

**Comment by vikas-cis** [comment]
@HenkSwardt  We can close and grab the block to txcount change from the above branch and which is update in the new PR #854

**Comment by kewde** [comment]
Electron notifications are broken, this is a potential replacement.
The default electron notification can cause application crashes.

**Comment by pciavald** [comment]
`electron-notify` does not work on Linux.
`electron-native-notification` was used as a replacement.

**Comment by vikas-cis** [comment]
@pciavald yeah I am checking on mac and it look something like this on dev 
screen shot 2018-02-23 at 3 56 18 pm

**Comment by vikas-cis** [comment]
@pciavald not working in mac build :( trying to debug it now

**Comment by vikas-cis** [comment]
@pciavald we can merge this as it looks great :)

**Comment by pciavald** [comment]
@vikas-cis @anandsinghparihar we need a test on the windows packaged version

**Comment by vikas-cis** [comment]
@pciavald let me try to build on windows 

Thanks

**Comment by anandsinghparihar** [comment]
@pciavald I am checking it in windows 10

File: modules/notification/notification.js [hunk 1]
```diff
@@ -1,31 +1,21 @@
...
-            'icon': path.join(__dirname, 'src/assets/icons/notification.png')
-        })
-        notification.show()
-        return Observable.create(observer => {
-            observer.complete(true);
-        });
-    });
+  rxIpc.registerListener('notification', (title, desc, params) => {
+    params = params ? params : { }; /* create param object if no param was given */
+    params.icon = '../../src/assets/icons/notification.png';
```
Comment (pciavald) [inline modules/notification/notification.js]: relative path does not work in linux packaged version

Comment (vikas-cis) [reply modules/notification/notification.js]: `src/assets/icons/notification.png` Is this work for linux ?

**Comment by vikas-cis** [comment]
@pciavald  I have tested on linux and it does take the absolute path but still it is not taking the image 
`[15:21:45.670] [debug] directory path... /home/cis/Desktop/partgui/packages/linux-unpacked/resources/app.asar
[15:21:45.671] [debug] directory path... /home/cis/Desktop/partgui/packages/linux-unpacked/resources/app.asar/src/assets/icons/notification.png`

**Comment by vikas-cis** [comment]
@anandsinghparihar can you please test now ?

Thanks

**Comment by vikas-cis** [comment]
@pciavald I have tested on linux as well Suyash system and it works well now :)

Thanks

**Comment by vikas-cis** [comment]
@pciavald works nicely on windows too :)
screen shot 2018-02-26 at 8 38 29 pm

**Comment by kewde** [comment]
@vikas-cis @anandsinghparihar 

Can we check if it works on OSX?
Thanks.

**Comment by vikas-cis** [comment]
@kewde yes it does works on OSX :)

Thanks

**Pull request merged**